charts/agent/templates/_helpers.tpl

```diff
@@ -333,26 +376,33 @@ secure.enabled=false we need to explicitly put those config entires in the
 agent config to prevent a backend push from enabling them after installation.
 */}}
 {{- define "agent.secureFeatures" }}
-    {{- $secureBlockConfig := dict "security" (dict
-        "enabled" .Values.secure.enabled
-        "k8s_audit_server_enabled" .Values.auditLog.enabled) }}
+    {{- if .Values.auditLog.enabled }}
+        {{- if or (and (hasKey .Values.global.sysdig "secure")
+                       (not .Values.global.sysdig.secure))
+                   .Values.secureLight }}
+            {{- $_ := set .Values.auditLog "enabled" false }}
+        {{- end }}
+    {{- end }}
+    {{- $secureConfig := dict "security" (dict "enabled" .Values.secure.enabled
+                                               "k8s_audit_server_enabled" .Values.auditLog.enabled) }}
     {{- if .Values.auditLog.enabled }}
         {{- range $key, $val := (dict
-                 "k8s_audit_server_url" .Values.auditLog.auditServerUrl
-                 "k8s_audit_server_port" .Values.auditLog.auditServerPort) }}
-            {{- $_ := set $secureBlockConfig.security $key $val }}
+             "k8s_audit_server_url" .Values.auditLog.auditServerUrl
+             "k8s_audit_server_port" .Values.auditLog.auditServerPort) }}
+        {{- $_ := set $secureConfig.security $key $val }}
         {{- end }}
     {{- end }}
-    {{- if not .Values.secure.enabled }}
+    {{- if or (not .Values.secure.enabled)
```

Comment:
I'm a bit puzzled about checking this vs checking global.sysdig, could you please elaborate it a bit?